sympy/solvers/diophantine.py

```diff
@@ -3182,7 +3182,7 @@ def power_representation(n, p, k, zeros=False):
             '''Todd G. Will, "When Is n^2 a Sum of k Squares?", [online].
                 Available: https://www.maa.org/sites/default/files/Will-MMz-201037918.pdf'''
             return
-        if feasible is 1:  # it's prime and k == 2
+        if not isinstance(feasible, bool) and feasible == 1:  # it's prime and k == 2
```

Comment:
Is feasible sometimes a bool?

Reply:
The `_can_do_sum_of_squares` returns True, False or 1. So the test of 'is 1' could only mean that the author was looking for 1. But `feasible==1` will mean that True would also pass. Since we know that `feasible` is True or 1 at this point in the code, the test `feasible is not True` is the simplest test to see that it is 1.